FRAuth/FRAuth/Config/OAuth2Client.swift

```diff
@@ -207,8 +207,12 @@ public class OAuth2Client: NSObject, Codable {
                     
                     let redirectURL = URL(string: redirectURLAsString)
                     
+                    // Verify that the response state value is the same
+                    if let state = redirectURL?.valueOf("state"), state != pkce.state {
```

Comment:
It is weird that `state` is part of the `pkce` object.
I don't think we should address this in this PR since it has been like that for a very long time.
The `state` is part of the [OAuth2 spec](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.2), and PKCE is another [spec](https://datatracker.ietf.org/doc/html/rfc7636)... 
Let's discuss this on Friday's meeting...